src/graphql/root/query/btc-price-list.ts

```diff
@@ -0,0 +1,87 @@
+import moment from "moment"
+import { GT } from "@graphql/index"
+
+import { MS_PER_DAY } from "@config/app"
+
+import PricePoint from "@graphql/types/object/price-point"
+import PriceGraphRange, {
+  priceRangeValues,
+} from "@graphql/types/scalar/price-graph-range"
+
+import { getHourlyPrice } from "@services/local-cache"
+import { getCurrentPrice } from "@services/realtime-price"
+
+const rangeStartDateMS: (string: typeof priceRangeValues[number]) => number = (range) => {
+  const currentTimestamp = Date.now()
+
+  switch (range) {
+    case "ONE_DAY":
+      return currentTimestamp - MS_PER_DAY
+    case "ONE_WEEK":
+      return currentTimestamp - 7 * MS_PER_DAY
+    case "ONE_MONTH":
+      return currentTimestamp - 30 * MS_PER_DAY
+    case "ONE_YEAR":
+      return currentTimestamp - 365 * MS_PER_DAY
+    case "FIVE_YEARS":
+      return currentTimestamp - 5 * 365 * MS_PER_DAY
+  }
+}
+
+const BtcPriceListQuery = GT.Field({
+  type: GT.List(PricePoint),
+  args: {
+    range: {
+      type: GT.NonNull(PriceGraphRange),
+    },
+  },
+  resolve: async (_, args, { logger }) => {
+    const { range } = args
+
+    const rangeStart = rangeStartDateMS(range)
+
+    if (!rangeStart) {
+      throw new Error("Invalid range")
+    }
+
+    const hourlyPrices = await getHourlyPrice({ logger })
+
+    const prices: PricePointType[] = []
+
+    for (const price of hourlyPrices) {
+      if (1000 * price.id < rangeStart) {
+        break
+      }
+      const btcPriceInCents = price.o * 100 * 10 ** 8
+      prices.push({
+        timestamp: price.id,
+        price: {
+          formattedAmount: btcPriceInCents.toString(),
+          base: Math.round(btcPriceInCents * 10 ** 4),
+          offset: 4,
+          currencyUnit: "USDCENT",
+        },
+      })
+    }
```

Comment:
this should be a reduce (or just an splice like in old server) also.. do we have 5 years of hourly prices in cache? (I dont see the filter in getHourlyPrice)

Reply:
I think `for...of` with early `break` here is better until we have an option to query the price data by date.